streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3630,6 +3768,20 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
 
   /**
    * Creates a stream from a subscription to a hub.
+   *
+   * {{{
+   * for {
+   *  hub   <- ZHub.unbounded[Chunk[Int]]
+   *  stream = ZStream.fromChunkHub(hub)
+   *  fiber <- stream.foreach(i => putStrLn(i.toString)).fork
+   *  _     <- clock.sleep(1.seconds)
+   *  _     <- hub.publish(Chunk(1, 2, 3))
+   *  _     <- fiber.join
+   * } yield ()
+   * // 1
+   * // 2
+   * // 3
+   * }}}
```

Comment:
I think it would be better to write this example using a latch:

```scala
for {
  promise <- Promise.make[Nothing, Unit]
  hub <- ZHub.unbounded[Chunk[Int]]
  stream = ZStream
    .fromChunkHub(hub)
    .ensuring(
      promise.succeed(())
    )
  fiber <- stream.foreach(i => putStrLn(i.toString)).fork
  _ <- promise.await
  _ <- hub.publish(Chunk(1, 2, 3))
  _ <- fiber.join
} yield ()
```